homeassistant/components/media_player/russound_rnet.py

```diff
@@ -91,33 +93,54 @@ def name(self):
 
     @property
     def state(self):
-        """Return the state of the device."""
-        return self._state
+        """Return the state of the device. directly from the device."""
+        if self._russ.get_power('1', self._zone_id) == 0:
+            return STATE_OFF
+        return STATE_ON
 
     @property
     def supported_media_commands(self):
         """Flag of media commands that are supported."""
         return SUPPORT_RUSSOUND
 
     @property
+    def source(self):
+        """Get the currently selected source."""
+        # Returns 0 based index for source.
+        index = self._russ.get_source('1', self._zone_id)
+        # Possibility exists that user has defined list of all sources.
+        # If a source is set externally that is beyond the defined list then
+        # an exception will be thrown.
+        # In this case fore it to the first element in sources.
+        try:
+            return self._sources[index]
+        except IndexError:
+            return self._sources[0]
```

Comment:
Returning `None` here might be better, to indicate that we don't know what source is selected.